### PR TITLE
feat(orchestrator)!: migrate to New Backend and Fix Deprecations

### PR DESCRIPTION
**Pull request by mareklibra**
Fixes: [RHIDP-4230](https://issues.redhat.com//browse/RHIDP-4230)

With this patch, the `orchestrator-backend` still loads well in the backstage of this `backstage-plugins` project on the `main` branch.

When the  `orchestrator-backend-dynamic` is exported and loaded to the `backstage-showcase` of the `main` branch, the plugin loads without any error and serves the requests as expected.

### CODE REVIEW

File: plugins/orchestrator-backend/src/service/ScaffolderService.ts [hunk 1]
```diff
@@ -1,7 +1,10 @@
-import { loggerToWinstonLogger, UrlReader } from '@backstage/backend-common';
-import { LoggerService } from '@backstage/backend-plugin-api';
-import { CatalogApi } from '@backstage/catalog-client';
-import { Config } from '@backstage/config';
+import { loggerToWinstonLogger } from '@backstage/backend-common';
```
Comment (mareklibra) [inline plugins/orchestrator-backend/src/service/ScaffolderService.ts:1]: This is the only reference `@backstage/backend-common`. By removing it, we can get rid of the dependency in the `package.json`. 
Still looking for the right way how to do it.

**Comment by mareklibra** [comment]
Cc @schultzp2020

**Comment by batzionb** [comment]
Seems like the workflow viewer is broken with this PR
It requires authentication, previously the authentication requirement was skipped
image

**Review by ydayagi — APPROVED** [review]

File: plugins/orchestrator-backend/src/service/ScaffolderService.ts (hunk 1 above)
Comment (AndrienkoAleksandr) [reply plugins/orchestrator-backend/src/service/ScaffolderService.ts:1]: I thought We were moving away from Winston logger in our plugins. For example: https://github.com/janus-idp/backstage-plugins/pull/1756/files

**Comment by mareklibra** [comment]
> It requires authentication, previously the authentication requirement was skipped

Thanks for spotting it. Newly skipping authentication in the `plugin.ts`.

File: plugins/orchestrator-backend/src/plugin.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+      },
+      async init(props) {
+        const { http } = props;
+        const router = await createRouter(props);
+        http.use(router);
+        http.addAuthPolicy({
+          path: '/ping',
+          allow: 'unauthenticated',
+        });
+        http.addAuthPolicy({
```
Comment (batzionb) [inline plugins/orchestrator-backend/src/plugin.ts:33]: please add skipping /docs as well

File: plugins/orchestrator-backend/src/plugin.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        scheduler: coreServices.scheduler,
+        auth: coreServices.auth,
+        httpAuth: coreServices.httpAuth,
+        http: coreServices.httpRouter,
+      },
+      async init(props) {
+        const { http } = props;
+        const router = await createRouter(props);
+        http.use(router);
+        http.addAuthPolicy({
```
Comment (batzionb) [inline plugins/orchestrator-backend/src/plugin.ts:29]: it used to be called /health
is there a reason it changed to /ping?

Comment (mareklibra) [reply plugins/orchestrator-backend/src/plugin.ts:29]: nope, mistake

Comment (mareklibra) [reply plugins/orchestrator-backend/src/plugin.ts:29]: fixed

File: plugins/orchestrator-backend/src/service/ScaffolderService.ts (hunk 1 above)
Comment (schultzp2020) [reply plugins/orchestrator-backend/src/service/ScaffolderService.ts:1]: Let's stick with your approach for now until https://github.com/backstage/backstage/issues/26933 is resolved.

**Review by schultzp2020 — APPROVED** [review]
/approve
/lgtm

**Pull request merged**